src/com/content-event/event-idea.js

```diff
@@ -30,35 +29,34 @@ export default class ContentEventIdea extends Component {
 	componentDidMount() {
 		$ThemeIdea.GetMy([this.props.node.id])
 		.then(r => {
+			//console.log(r);
 			if ( r.ideas ) {
-				this.setState({"ideas": r.ideas});
+				this.setState({'ideas': r.ideas, 'enableSubmit': canHaveMoreIdeas(r.ideas)});
 			}
 			else {
-				this.setState({"ideas": {}});
+				this.setState({'ideas': {}});
 			}
 		})
 		.catch(err => {
-			this.setState({"error": err});
+			this.setState({'error': 'Error fetching your previous suggestions. Make sure you are logged in.'});
 		});
 	}
 
-//		<script>
-//			document.getElementById("input-idea").addEventListener("keydown", function(e) {
-//				if (!e) { var e = window.event; }
-//				if (e.keyCode == 13) { /*e.preventDefault();*/ submitIdeaForm(); }
-//			}, false);
-//		</script>
-
-	textChange( e ) {
-		this.setState({"idea": e.target.value.trim()});
+	textChange( e, isSubmit ) {
+		let idea = e.target.value;
+		if (isSubmit) {
```

Comment:
Prefer:

```js
if ( isSubmit ) {
```

Another style thing I can't convince eslint to enforce. 

The idea is that only function calls don't have spaces between brackets. Function declarations do however. This way you can tell the difference between a function declaration and a call just by glancing.

```js
myFunction( e ) {                 // declaration
        myFunction("something");  // call
}
```